File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
             List<Integer> l=new ArrayList<Integer> ();
        
        traversal(l,root);
        return l;
        
    }
    
    public void traversal(List<Integer> l , TreeNode root)
    {
        if(root==null)
            
            return;
         
        traversal(l, root.left);
        traversal(l,root.right);
        l.add(root.val);       

        
    }
}

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        li=[]
        self.traversal(root,li)
        return li
    def traversal(self,root,li):
        if root is None:
            return
        self.traversal(root.left,li)
        self.traversal(root.right,li)
        li.append(root.val)
        